platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/address/AddressBook.java

```diff
@@ -423,24 +440,26 @@ private void updateAddress(@NonNull final Address address) {
     }
 
     /**
-     * Add a new address.
+     * Add a new address at the end of the address book.
      *
      * @param address the address to add
      */
     private void addNewAddress(@NonNull final Address address) {
         final NodeId addressNodeId = address.getNodeId();
-        final int nodeIdComparison = addressNodeId.compareTo(nextNodeId);
+        final int addressBookSize = getSize();
+        final NodeId nextAvailable = addressBookSize == 0
+                ? NodeId.FIRST_NODE_ID
+                : getNodeId(getSize() - 1).getOffset(1);
```

Comment:
I am wondering if this nextAvailable is correct.

Reply:
We discussed on a zoom call, this is only preventing the insert of a new address between existing address and not trying to to model `nextNodeId` with multiple deletes at the end of a previous address book.